Add customDirectory for diagnostics

xylar opened the pull request:
This allows a user to supply their own observations, mapping files or region masks (without putting them in a shared diagnostics directory). This is appropriate when working with custom meshes or testing new observations or regions that aren't (yet) standard in MPAS-Analysis.

In the process of testing, I found bugs in `timeSeriesOceanRegions` that I have fixed here.  They relate to the names of subtasks, which were not unique when a second region group is defined.

closes #604

pep8speaks commented:
Hello @xylar! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:











* In the file [`mpas_analysis/shared/io/utility.py`](https://github.com/MPAS-Dev/MPAS-Analysis/blob/9a674717bae1081636317b20b6aa9b9644c21c69/mpas_analysis/shared/io/utility.py):

> [Line 204:39](https://github.com/MPAS-Dev/MPAS-Analysis/blob/9a674717bae1081636317b20b6aa9b9644c21c69/mpas_analysis/shared/io/utility.py#L204): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent

xylar commented:
## Testing

I tested this branch with the following:
* I set `customDirectory` to point to the location of the full set of diagnostics data and left `baseDirectory` pointing to the dummy value (`path/to/diagnostics`), and successfully ran QU240wLI analysis
* I created a `MediterraneanBasins.geojson` file with the Mediterranean as a feature and put it in `custom_diags/mpas_analysis/region_masks`.  I added the full path to this directory to my config file.  I also added a new `RegionsGroup` to `timeSeriesOceanRegions` pointing to the geojson file (just the file name, not the full path).  The analysis successfully ran (after some bug fixes listed above), creating a new region mask and plotting time series of Mediterranean mean T, S and potential density.
* I successfully ran my usual QU240wLI analysis in python 2.7 and 3.7.
